### ISSUE TITLE
java.lang.AssertionError: assertion failed

### ISSUE THREAD

**Issue by rfkm** [coursier/coursier#589]
I was thinking #169 was fixed by #567, but I've noticed the latest version (1.0.0-RC5) of coursier still fails to resolve `org.jitsi:jitsi-videobridge:1.0-SNAPSHOT` due to another error:

```
Exception in thread "main" java.lang.AssertionError: assertion failed
        at scala.Predef$.assert(Predef.scala:156)
        at coursier.TermDisplay$UpdateDisplayRunnable.newEntry(TermDisplay.scala:213)
        at coursier.TermDisplay.downloadingArtifact(TermDisplay.scala:492)
        at coursier.Cache$$anonfun$remote$1$1$$anonfun$apply$26.apply(Cache.scala:669)
        at coursier.Cache$$anonfun$remote$1$1$$anonfun$apply$26.apply(Cache.scala:669)
        at scala.Option.foreach(Option.scala:257)
        at coursier.Cache$$anonfun$remote$1$1.apply(Cache.scala:669)
        at coursier.Cache$$anonfun$remote$1$1.apply(Cache.scala:556)
        at scalaz.concurrent.Task$.Try(Task.scala:457)
        at scalaz.concurrent.Task$$anonfun$apply$25.apply(Task.scala:354)
        at scalaz.concurrent.Task$$anonfun$apply$25.apply(Task.scala:354)
        at scalaz.concurrent.Future$$anonfun$apply$15$$anon$3.call(Future.scala:432)
        at scalaz.concurrent.Future$$anonfun$apply$15$$anon$3.call(Future.scala:432)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:748)
```
In addition, the progress indicator looks strange:
![coursier-glitches](https://user-images.githubusercontent.com/857162/27390450-39e9627c-56dc-11e7-91a7-457bd2105b28.png)

I've never seen such error during working on #567, but it seems to occur only with snapshot dependencies (as far as I noticed), so my changes may directly or indirectly relate to this problem.

**Comment by alexarchambault** [comment]
It has to see with two dependencies (e.g. `org.jitsi:jitsi-universe:1.0-SNAPSHOT` and `org.jitsi:jitsi-universe:1.0-20160405.235512-16`), that try to download the same stuff (`https://github.com/jitsi/jitsi-maven-repository/raw/master/snapshots/org/jitsi/jitsi-universe/1.0-SNAPSHOT/`) in the same iteration of the resolution, which the API of `coursier.Cache.Logger` doesn't really allow... (it ~uniquely identifies downloads via their URL).

I have a fix for that locally, that I'll push after some clean-up...

**Comment by rfkm** [comment]
Ah, that makes sense. Thanks!

**Comment by rfkm** [comment]
oops, mistouched..

**Comment by alexarchambault** [comment]
https://github.com/coursier/coursier/commit/d437cfb87d0f4b5ddad4a46a8282dc3830557b18 fixes that for me. It will be included in `1.0.0-RC6` about to be released. (Re-open if ever you still run into this...)

**Comment by rfkm** [comment]
@alexarchambault 
I cannot show you a reliable repro, but this issue still sometimes happens...
I feel it frequently occurs when Coursier detects a newer snapshot artifact.

stacktrace (1.0.0-RC8):
```
Caused by: java.lang.AssertionError: assertion failed
        at scala.Predef$.assert(Predef.scala:165)
        at coursier.TermDisplay$UpdateDisplayRunnable.newEntry(TermDisplay.scala:166)
        at coursier.TermDisplay.downloadingArtifact(TermDisplay.scala:445)
        at coursier.Cache$$anonfun$remote$1$1$$anonfun$apply$26.apply(Cache.scala:669)
        at coursier.Cache$$anonfun$remote$1$1$$anonfun$apply$26.apply(Cache.scala:669)
        at scala.Option.foreach(Option.scala:236)
        at coursier.Cache$$anonfun$remote$1$1.apply(Cache.scala:669)
        at coursier.Cache$$anonfun$remote$1$1.apply(Cache.scala:556)
        at scalaz.concurrent.Task$.Try(Task.scala:457)
        at scalaz.concurrent.Task$$anonfun$apply$25.apply(Task.scala:354)
        at scalaz.concurrent.Task$$anonfun$apply$25.apply(Task.scala:354)
        at scalaz.concurrent.Future$$anonfun$apply$15$$anon$3.call(Future.scala:432)
        at scalaz.concurrent.Future$$anonfun$apply$15$$anon$3.call(Future.scala:432)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:748)
```
